compiler-core/src/exhaustiveness.rs

```diff
@@ -1178,11 +1211,55 @@ pub enum BitArrayMatchedValue {
     },
 }
 
+impl PartialEq for BitArrayMatchedValue {
```

Comment:
Does this need to be a trait implementation? We always try to avoid traits in favour of regular methods, to keep the codebase as explicit and as easy to understand as possible 🙏